corehq/ex-submodules/phonelog/tasks.py

```diff
@@ -17,3 +23,31 @@ def purge_old_device_report_entries():
     UserErrorEntry.objects.filter(server_date__lt=max_age).delete()
     ForceCloseEntry.objects.filter(server_date__lt=max_age).delete()
     UserEntry.objects.filter(server_date__lt=max_age).delete()
+
+
+@task
+def send_device_logs_to_sumologic(domain, xform):
+    url = getattr(settings, 'SUMOLOGIC_URL', None)
+    if url and SUMOLOGIC_LOGS.enabled(xform.form_data.get('device_id'), NAMESPACE_OTHER):
+        for fmt in ['log', 'user_error', 'force_close']:
+            headers = {"X-Sumo-Category": "{env}/{domain}/{fmt}".format(
+                env=_get_sumologic_environment(),
+                domain=domain,
+                fmt=fmt,
+            )}
+            data = getattr(SumoLogicLog(domain, xform), "{}_subreport".format(fmt))()
```

Comment:
This kind of function / property access makes it susceptible to bugs later on if anyone refactors / moves the functions around since they won't show up in a search for the function name.

Reply:
:+1: